### ISSUE TITLE
Discrepancy between content and media URL when containing special characters

### ISSUE THREAD

**Issue by ane-dwarf** [umbraco/Umbraco-CMS#15939]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

12.3.7

### Bug summary

Creating content that has a name containing special characters (eg. "ÅÅ Øø Ææ Ã") will create a URL that matches the request handler settings but uploading media with a filename that contains special characters will not follow the request handler settings.

### Specifics

```json
"RequestHandler": {
    "EnableDefaultCharReplacements": false
}
```

The settings above will generate the following URLs:

![image](https://github.com/umbraco/Umbraco-CMS/assets/37106288/1d390c8c-b499-4fbb-944e-8a1681386173)

![image](https://github.com/umbraco/Umbraco-CMS/assets/37106288/a983f623-40dd-4c3a-9835-06f06cf95e0b)

```json
"RequestHandler": {
    "ConvertUrlsToAscii": "try",
    "EnableDefaultCharReplacements": true,
    "UserDefinedCharCollection": [
        {
            "Char": "Å",
            "Replacement": "first"
        },
        {
            "Char": "Ø",
            "Replacement": "second"
        },
        {
            "Char": "Æ",
            "Replacement": "third"
        }
    ]
}
```

The settings above will create the following URLs:

![image](https://github.com/umbraco/Umbraco-CMS/assets/37106288/d82b95aa-bdeb-46fc-b072-46f628582e3c)

![image](https://github.com/umbraco/Umbraco-CMS/assets/37106288/e3142450-d7f2-4e7b-ab10-29e8bb33443a)

### Steps to reproduce

Upload media with special characters in the filename setting `EnableDefaultCharReplacements` to both `true` and `false`. Create content that has the same special characters in the name.

### Expected result / actual result

I would expect the same replacements to be made for both media and content in the URL.

---
_This item has been added to our backlog AB#42359_

**Comment by Migaroez** [comment]
Hey @ane-dwarf I managed to partially reproduce your issue on 13.4.0
and with even stranger behavior in 14.1.0-rc2

For clarification as the Images are very hard to compare
Document Name: `ÅÅ Øø Ææ Ã`
Document url with default settings: `/aaaa-oeoe-aeae-a/`
Document url with supplied settings: `/firstfirst-secondoe-thirdae-a/`

Uploaded File name: `ÅÅ Øø Ææ Ã.svg`
Media url with default settings: `aaaa-oeoe-aeae-ã.svg`
Media url with supplied settings[v13]: `firstfirst-secondoe-thirdae-ã.svg`
Media url with supplied settings[v14]: `aaaa-oeoe-aeae-ã.svg`

**Comment by Migaroez** [comment]
he @ane-dwarf, while we look into this, have you tried to include both lower case and uppercase characters inside the `UserDefinedCharCollection` to solve your needs?

**Comment by ane-dwarf** [comment]
Hi @Migaroez, no I haven't worked with this since I posted the issue so it's kinda just laying around until it's picked up again (when there's time 😄). I'll let you know if I get a chance to look at it.

**Comment by kjac** [comment]
Fix in #17580
